### PR TITLE
Remove lookups from the layers

### PR DESCRIPTION
**Pull request by kchoo1118**

### CODE REVIEW

**Comment by kchoo1118** [comment]
The conv is not fixed yet.
I think maybe I will just change it to a hypercube only conv net

File: NetKet/Machine/abstract_layer.hpp [hunk 1]
```diff
@@ -79,18 +79,6 @@ class AbstractLayer {
   @param sigma is the variance of the gaussian.
   */
   virtual void InitRandomPars(int seed, double sigma) = 0;
-  /**
-  Member function to initialise the lookup for the later. "Summary: This
```
Comment (gcarleo) [inline NetKet/Machine/abstract_layer.hpp]: Why remove the comment? :)

File: NetKet/Machine/abstract_layer.hpp [hunk 2]
```diff
@@ -79,18 +79,6 @@ class AbstractLayer {
...
-  function uses the input vector v to compute the lookup lt and the output. This
-  function also resizes the lookup elements to the correct size."
-  @param v is the input vector.
-  @param theta is a vector in the lookuptable storing the output of the
-  layer before the nonlinear function is applied.
-  @param output is a vector in the lookuptable storing the previous output of
-  the current layer.
-  */
-  virtual void InitLookup(const VectorType &v, LookupType &lt,
```
Comment (gcarleo) [inline NetKet/Machine/abstract_layer.hpp]: Finally gone!

File: NetKet/Machine/abstract_layer.hpp [hunk 3]
```diff
@@ -114,7 +102,7 @@ class AbstractLayer {
   */
   virtual void UpdateLookup(const VectorType &input,
```
Comment (gcarleo) [inline NetKet/Machine/abstract_layer.hpp]: Maybe we can just call these UpdateOutput, just to remove any reference to Lookup tables ?

File: NetKet/Machine/abstract_layer.hpp [hunk 4]
```diff
@@ -185,7 +157,6 @@ class AbstractLayer {
   */
   virtual void Backprop(const VectorType &prev_layer_output,
                         const VectorType &this_layer_output,
-                        const LookupType &this_layer_theta,
```
Comment (gcarleo) [inline NetKet/Machine/abstract_layer.hpp]: Great

File: NetKet/Machine/sum_output.hpp [hunk 5]
```diff
@@ -84,30 +84,20 @@ class SumOutput : public AbstractLayer<T> {
 
   void SetParameters(VectorConstRefType /*pars*/, int /*start_idx*/) override {}
```
Comment (gcarleo) [inline NetKet/Machine/sum_output.hpp]: I never liked these SetParameters and GetParameters with two arguments... Can't we just use an Eigen:Map as you have done in other parts? https://eigen.tuxfamily.org/dox/group__TutorialReshapeSlicing.html

**Comment by gcarleo** [comment]
@kchoo1118  the easiest is to use https://eigen.tuxfamily.org/dox/group__TutorialBlockOperations.html 

There is the very nice 
```c++
vector.segment(i,n);
```

File: NetKet/Machine/abstract_layer.hpp (hunk 3 above)
Comment (kchoo1118) [reply NetKet/Machine/abstract_layer.hpp]: sure.

File: NetKet/Machine/sum_output.hpp (hunk 5 above)
Comment (kchoo1118) [reply NetKet/Machine/sum_output.hpp]: sure that is possible.

**Pull request merged**